firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -613,6 +613,28 @@ class BrowserViewController: UIViewController,
         }
     }
 
+    private func updateAddressToolbarContainerPosition(for traitCollection: UITraitCollection) {
+        guard searchBarPosition == .bottom, isToolbarRefactorEnabled else { return }
+
+        let isNavToolbar = toolbarHelper.shouldShowNavigationToolbar(for: traitCollection)
+        header.removeArrangedView(addressToolbarContainer, animated: false)
+        overKeyboardContainer.removeArrangedView(addressToolbarContainer, animated: false)
+
+        if isNavToolbar {
+            overKeyboardContainer.addArrangedSubview(addressToolbarContainer)
+        } else {
+            header.addArrangedSubview(addressToolbarContainer)
+        }
+        updateHeaderConstraints()
+
+        let action = GeneralBrowserMiddlewareAction(
+            scrollOffset: scrollController.contentOffset,
+            toolbarPosition: isNavToolbar ? searchBarPosition : .top,
+            windowUUID: windowUUID,
+            actionType: GeneralBrowserMiddlewareActionType.toolbarPositionChanged)
+        store.dispatchLegacy(action)
+    }
```

Comment:
We should leverage existing code to avoid causing side effects or new bugs. 
```suggestion
    private func updateAddressToolbarContainerPosition(for traitCollection: UITraitCollection) {
        guard searchBarPosition == .bottom, isToolbarRefactorEnabled else { return }

        let isNavToolbar = toolbarHelper.shouldShowNavigationToolbar(for: traitCollection)
        let newPosition: SearchBarPosition = isNavToolbar ? .bottom : .top
        let notificationObject = [PrefsKeys.FeatureFlags.SearchBarPosition: newPosition]
        notificationCenter.post(name: .SearchBarPositionDidChange, withObject: notificationObject)
    }
```